warcry_proj/warcry_proj/views/default.py

```diff
@@ -0,0 +1,139 @@
+from pyramid.httpexceptions import HTTPOk
+import transaction
+import datetime
+import random
+import json
+from json import JSONDecoder
+from json import JSONEncoder
+from datetime import datetime
+from sqlalchemy import func
+from sqlalchemy import desc
+from pyramid.response import Response
+
+from pyramid.view import view_config
+
+from warcry_proj.models.mymodel import (
+    DBSession,
+    User,
+    UserSchema,
+    )
+
+class DateTimeDecoder(json.JSONDecoder):
+    def __init__(self, *args, **kargs):
+        JSONDecoder.__init__(self, object_hook=self.dict_to_object,*args, **kargs)
+
+    def dict_to_object(self, d):
+        if '__type__' not in d:
+            return d
+
+        type = d.pop('__type__')
+        try:
+            dateobj = datetime(**d)
+            return dateobj
+        except:
+            d['__type__'] = type
+            return d
+
+class DateTimeEncoder(JSONEncoder):
+    """ Instead of letting the default encoder convert datetime to string,
+        convert datetime objects into a dict, which can be decoded by the
+        DateTimeDecoder
+    """
+
+    def default(self, obj):
+        if isinstance(obj, datetime):
+            return {
+                '__type__' : 'datetime',
+                'year' : obj.year,
+                'month' : obj.month,
+                'day' : obj.day,
+                'hour' : obj.hour,
+                'minute' : obj.minute,
+                'second' : obj.second,
+                'microsecond' : obj.microsecond,
+            }
+        else:
+            return JSONEncoder.default(self, obj)
+
+#Get a list of all Users in Database
+@view_config(route_name='all_users', request_method='GET', renderer='json')
+def get_all_users(_):
+    schema = UserSchema(many=True)
+    return schema.dumps(DBSession.query(User).all(), cls=DateTimeEncoder)
+
+#Get a single user by specifying the user ID
+@view_config(route_name='single_user', request_method='GET', renderer='json')
```

Comment:
How does the system handle parameter validation? Does it inherently protect against sql injection as part of the ORM? (Usually they do, but I wanted to double check). How do you make sure for example, that maybe a username has to be all lowercase with no spaces, or that it can't contain an apostrophe? I'm not saying you have to add full validation examples, but one might be helpful just to show it working.